### PR TITLE
`vms/platformvm`: Add `processStandardTxs` helper

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

Adds a `processStandardTxs` helper which will also be used in `ProposalBlock`s

Factored out of https://github.com/ava-labs/avalanchego/pull/2451

## How this works

pretty straightforward

## How this was tested

CI

### CODE REVIEW

**Review by StephenButtolph — APPROVED** [review]

File: vms/platformvm/block/executor/verifier.go [hunk 1]
```diff
@@ -394,43 +395,67 @@ func (v *verifier) standardBlock(
...
 		timestamp:      onAcceptState.GetTimestamp(),
-		atomicRequests: make(map[ids.ID]*atomic.Requests),
+		inputs:         inputs,
+		atomicRequests: atomicRequests,
 	}
 
-	// Finally we process the transactions
-	funcs := make([]func(), 0, len(b.Transactions))
-	for _, tx := range b.Transactions {
+	v.Mempool.Remove(b.Transactions)
```
Comment (patrick-ogrady) [inline vms/platformvm/block/executor/verifier.go:415]: nit: I know this doesn't matter when everything is behind a single lock, but it feels better to me to remove transactions from the mempool before we add to the `blkIDToState` map (which is when other processes could conceivably query this info, assuming that all related txs are no longer in the mempool).

File: vms/platformvm/block/executor/verifier.go [hunk 2]
```diff
@@ -394,43 +395,67 @@ func (v *verifier) standardBlock(
...
 
 		for chainID, txRequests := range txExecutor.AtomicRequests {
 			// Add/merge in the atomic requests represented by [tx]
-			chainRequests, exists := blkState.atomicRequests[chainID]
+			chainRequests, exists := atomicRequests[chainID]
 			if !exists {
-				blkState.atomicRequests[chainID] = txRequests
+				atomicRequests[chainID] = txRequests
 				continue
 			}
```
Comment (patrick-ogrady) [inline vms/platformvm/block/executor/verifier.go:461]: I'm assuming we assert there are no overlapping requests elsewhere?

**Review by abi87 — COMMENTED** [review]
I'd defer merging this after having evaluated the alternative design in https://github.com/ava-labs/avalanchego/pull/2439 and https://github.com/ava-labs/avalanchego/pull/2442, which should be slightly more memory efficient albeit slightly more complex to review

File: vms/platformvm/block/executor/verifier.go (hunk 1 above)
Comment (dhrubabasu) [reply vms/platformvm/block/executor/verifier.go]: done

File: vms/platformvm/block/executor/verifier.go (hunk 2 above)
Comment (dhrubabasu) [reply vms/platformvm/block/executor/verifier.go:462]: Correct, this is the only place we modify this map for the block

**Review by abi87 — APPROVED** [review]

**Pull request merged**